middlewares/retry.go

```diff
@@ -114,107 +100,69 @@ func (l RetryListeners) Retried(req *http.Request, attempt int) {
 	}
 }
 
-type retryResponseRecorder interface {
+type retryResponseWriter interface {
```

Comment:
why rename to `retryResponseRecorder` to `retryResponseWriter` and make a  `errorPagesResponseRecorder` ?

Reply:
I renamed it because the current implementation does not record anything anymore. It writes directly through to the original response writer in case the request should not be retried, so it's no recorder anymore.

Regarding the `errorPagesResponseRecorder`:

> As the former response recorder was used also at the error handler, I moved it there. I think the error handler is not that critical when it comes to response sizes, e.g. no files will be transferred through it I guess, but there is also room for improvement for the future.

It's basically that the `retryResponseRecorder` was re-used in the error pages middleware and that I didn't want to make changes there. As the error pages middleware is the last place where the original `retryResponseRecorder` is used I renamed it accordingly.